weaver/cli.py

```diff
@@ -242,6 +242,20 @@ class RequestAuthHandler(AuthHandler, HTTPBasicAuth):
     """
     Base class to send a request in order to retrieve an authorization token.
     """
+
+    def __init__(self, token=None, **kwargs):
+        # type: (Any, **Any) -> None
+        AuthHandler.__init__(self, **kwargs)
+        HTTPBasicAuth.__init__(self, username=kwargs.get("username"), password=kwargs.get("password"))
+        self.token = token
+
+    # Update __init__'s signature to include keyword arguments from AuthHandler's __init__. This is required because
+    # the CLI entrypoint inspects the auth class' signature in order to figure out which arguments to pass to it.
+    _signature_AuthHandler = inspect.signature(AuthHandler.__init__)
+    __init__.__signature__ = _signature_AuthHandler.replace(
+        parameters=(*_signature_AuthHandler.parameters.values(), inspect.signature(__init__).parameters["token"])
+    )
```

Comment:
Seems like it would be less hackish to simply add `username, password` parameters to `__init__` as in the parent class and pass them to child classes.

Reply:
So I'm willing to do that if you'd prefer. 
But I don't really see this as a hack especially because manipulating the `__signature__` is given as an example in [PEP362](https://peps.python.org/pep-0362/#visualizing-callable-objects-signature) and if we manually add the `username`, `password`, etc. then we lose the advantage of inheriting the signature from the parent class (if the parent class's signature changes it requires updating the child classes).

Reply:
I think the literal signature is more useful since it gives direct feedback to potential users of which parameters it takes. 
I'm not too worried about changes in the parent signature since IDEs can flag those issues (at least PyCharm does), and some tests will fail if the problem is important.